# 1. 下面两段代码输出的结果相同么？请解释原因

In [14]:
x1,y1=4,5
x2,y2='4','5'
x2,y2='4','5'
print(x1+y1)
print(x2+y2)

9
45


一是整数间加法运算,结果为 9 ；二是字符串之间的拼接操作，结果为 45。

# 2.  ‘10/3'，‘10//3’和'10%3'结果相同么？请说明原因

In [24]:
x=10/3
y=10//3
z=10%3
print(x,'\n',y,'\n',z)

3.3333333333333335 
 3 
 1


结果分别为：3.333  ，3  ，1。其中，/是除法运算、//所得结果向下取整、%是取余运算。

# 3. a = 3，str(a*3) + str(a)*3的输出是什么?


In [28]:
a=3
print(str(a*3))
print(str(a)*3)
print("输出为:",str(a*3)+str(a)*3)#加号为字符串拼接

9
333
输出为: 9333


# 4. my_string = 'this is my string', 如何查看这个字符串长度

In [2]:
my_string="this is my string"
print(len(my_string))

17


# 5. 使用 while 来计算 1 到 100 的总和

In [1]:
i=0
sum=0
while(i<101):
    sum=sum+i;
    i=i+1

print(sum)

5050


# 6. 输入一年份，判断该年份是否是闰年并输出结果

In [3]:
year=int(input('请输入一个整数'))
if(year%400==0 or year%4==0 and year%100!=0):
    print("是闰年")
else:
    print("不是闰年")

请输入一个整数2040
是闰年


# 7.使用while, 完成以下图形的输出

In [6]:
i="*"
j=0
k=5
while(j<5):
    j=j+1
    print(i*j)
while(k>0):
    k=k-1
    print(k*i)

*
**
***
****
*****
****
***
**
*



#  8. 输入两个数，比较大小后，从小到大升序打印

In [9]:
a=int(input("请输入第一个数"))
b=int(input("请输入第二个数"))
if(a<b):
    print(a,b)
else:
    print(b,a)


请输入第一个数7
请输入第二个数1
1 7


# 9. 有1、2、3、4个数字，能组成多少个互不相同且无重复数字的三位数？都是多少？

In [11]:
i=0
for a in range(1,5):
    for b in range(1,5):
        for c in range(1,5):
            if a!=b and a!=c and b!=c:
                print("%d%d%d"%(a,b,c))
                i+=1

123
124
132
134
142
143
213
214
231
234
241
243
312
314
321
324
341
342
412
413
421
423
431
432


# 10. 一个整数，它加上100后是一个完全平方数，再加上168又是一个完全平方数，请问该数是多少？

In [10]:
i=0
j=0
k=0
for i in range(1,200):
    for j in range(1, 200):
        for k in range(1,200):
            if(i+100==j*j and i+168==k*k):
                    print(i)


156


# 11. 求200以内能被17整除的最大正整数
## 例如：输入18，输出18=2*3*3

In [12]:
import random
x=int(input("输入一个正整数:"))
a=list()
for i in range(2,x+1):
    while 1:
        if x%i==0:
            a.append(i)
            x/=i
        else:
            break
flag=0
for i in a:
    if flag==0:
        print (i,end='')
        flag=1
    else:
        print('*',i,end='')

输入一个正整数:12
2* 2* 3

## 13. 一球从100米高度自由落下，每次落地后反跳回原高度的一半，再落下，求他在第10次落地时，共经过多少米？第10次反弹多高？

In [13]:
h=100
h_sum=100
for i in range(1,11):
    h=h/2
    h_sum+=h*2
print("共经过",h_sum,"米")
print("第十次反弹",h,"米")

共经过 299.8046875 米
第十次反弹 0.09765625 米


# 14. 编写代码，实现求100-200里面所有的素数

In [14]:
a=[]
for i in range(100,201):
    b=0
    for j in range(2,i-1):
        if i%j==0:
            b+=1
    if b==0:
        a.append(i)
print(a)

[101, 103, 107, 109, 113, 127, 131, 137, 139, 149, 151, 157, 163, 167, 173, 179, 181, 191, 193, 197, 199]


#  15. d={‘a’:1,’b’:2,’c’:3}请打印出key、value对

In [15]:
d={'a':1,'b':2,'c':3}
for key in d:
    print(key,d[key])

a 1
b 2
c 3
